src/BoundaryConditions/field_boundary_conditions.jl

```diff
@@ -14,47 +14,50 @@ Construct a `FieldBoundaryConditions` using a `CoordinateBoundaryCondition` for
 """
 FieldBoundaryConditions(x, y, z) = FieldBoundaryConditions((x, y, z))
 
-"""
-    HorizontallyPeriodicBCs(;   top = BoundaryCondition(Flux, nothing),
-                             bottom = BoundaryCondition(Flux, nothing))
-
-Construct `FieldBoundaryConditions` with `Periodic` boundary conditions in the x and y
-directions and specified `top` (+z) and `bottom` (-z) boundary conditions for u, v,
-and tracer fields.
-
-`HorizontallyPeriodicBCs` cannot be applied to the the vertical velocity w.
-"""
-function HorizontallyPeriodicBCs(;    top = BoundaryCondition(Flux, nothing),
-                                   bottom = BoundaryCondition(Flux, nothing))
-
-    x = PeriodicBCs()
-    y = PeriodicBCs()
-    z = CoordinateBoundaryConditions(bottom, top)
-
-    return FieldBoundaryConditions(x, y, z)
-end
-
-"""
-    ChannelBCs(; north = BoundaryCondition(Flux, nothing),
-                 south = BoundaryCondition(Flux, nothing),
-                   top = BoundaryCondition(Flux, nothing),
-                bottom = BoundaryCondition(Flux, nothing))
-
-Construct `FieldBoundaryConditions` with `Periodic` boundary conditions in the x
-direction and specified `north` (+y), `south` (-y), `top` (+z) and `bottom` (-z)
-boundary conditions for u, v, and tracer fields.
-
-`ChannelBCs` cannot be applied to the the vertical velocity w.
-"""
-function ChannelBCs(;  north = BoundaryCondition(Flux, nothing),
-                       south = BoundaryCondition(Flux, nothing),
-                         top = BoundaryCondition(Flux, nothing),
-                      bottom = BoundaryCondition(Flux, nothing)
-                    )
-
-    x = PeriodicBCs()
-    y = CoordinateBoundaryConditions(south, north)
-    z = CoordinateBoundaryConditions(bottom, top)
+default_bc(::Periodic) = PeriodicBC()
+default_bc(::Bounded)  = NoFluxBC()
+
+valid_directions = (:west, :east, :south, :north, :bottom, :top)
+
+function FieldBoundaryConditions(grid; kwargs...)
+    kws = keys(kwargs)
+    for kw in kws
+        if kw ∉ valid_directions
```

Comment:
Another strategy would be to set all defaults in the constructor to 'nothing', and then reset them to default if they are nothing. The error the user receives would then bean ordinary KeywordError. This also may simplify subsequent validation.

Reply:
I ended up assigning default boundary conditions in the constructor which cleaned up the function quite a bit.